core/include/gnuradio-4.0/Scheduler.hpp

```diff
@@ -153,19 +164,41 @@ class SchedulerBase {
     }
 
     void
-    processScheduledMessages() {
-        // Process messages in scheduler
-        auto passMessages = [](auto &inPort, auto &outPort) {
-            auto &reader = inPort.streamReader();
-            if (const auto available = reader.available(); available > 0) {
-                const auto &input = reader.get(available);
-                outPort.streamWriter().publish([&](auto &output) { std::ranges::copy(input, output.begin()); }, available);
+    processScheduledMessages(bool block) {
+        auto &msgInReader = msgIn.streamReader();
+        if (const auto available = msgInReader.available(); available > 0 || block) {
+            const auto  toRead = std::max(available, 1UZ);
+            const auto &input  = msgInReader.get(toRead);
+            for (const auto &msg : input) {
+                auto &toChildWriter = _toChildMessagePort.streamWriter();
+                const auto kindIt        = msg.find(gr::message::key::Kind);
+                // TODO factor out, protect against throw in std::get
+                const auto kind = kindIt != msg.end() ? std::get<std::string>(kindIt->second) : std::string{};
+                fmt::println("Received message. kind: {}", kind);
+                using namespace gr::message::scheduler::command;
+                if (kind == StartScheduler) {
+                    self().start();
```

Comment:
for multithreaded schedulers, this currently run on the worker threads, which isn't safe. Somehow need to make these thread-safe, or handle them on the scheduler's main thread.